### ISSUE TITLE
Better errors for missing Dict keys

### ISSUE THREAD

**Issue by rafaqz** [yeesian/ArchGDAL.jl#178]
It's possible to get errors like this when you try to create a file with the wrong data type, and the `_GDALTYPE` Dict is accessed:

```julia
ERROR: KeyError: key Int64 not found
Stacktrace:
 [1] getindex at ./dict.jl:467 [inlined]
 [2] #unsafe_create#56 at /root/.julia/packages/ArchGDAL/j9NPL/src/dataset.jl:200 [inlined]
 [3] create(::GeoData.var"#190#191"{GeoArray{Int64,3,Tuple{Lon{LinRange{Float64},Projected{
 ```

It could be better to throw a more descriptive error, like "`Int64` is not a recognised data type" etc. I'm not sure if this also applies to the other const Dicts in use like `_JLType`

See https://github.com/rafaqz/GeoData.jl/issues/143

**Comment by evetion** [comment]
I have similar issues over at https://github.com/evetion/GeoArrays.jl/issues/42, would be nice if we can solve it within ArchGDAL.

**Comment by yeesian** [comment]
Addressed by #179.
